packages/clay-form/src/SelectBox.tsx

```diff
@@ -0,0 +1,215 @@
+
+	return arrayToMove;
+}
+
+function reorderUp(array: Array<TItem>, selectedIndexes: Array<number>) {
+	let clonedArray = [...array];
+
+	for (let i = 0; i < selectedIndexes.length; i++) {
+		const item = selectedIndexes[i];
+
+		if (item === 0) {
+			return clonedArray;
+		}
+
+		clonedArray = arrayMove(clonedArray, item, item - 1);
+	}
+
+	return clonedArray;
+}
+
+function reorderDown(array: Array<TItem>, selectedIndexes: Array<number>) {
+	let clonedArray = [...array];
+
+	for (let i = 0; i < selectedIndexes.length; i++) {
+		const item = selectedIndexes[i];
+
+		if (selectedIndexes.includes(clonedArray.length - 1)) {
+			return clonedArray;
+		}
+
+		clonedArray = arrayMove(clonedArray, item, item + 1);
+	}
+
+	return clonedArray;
+}
+
+interface IProps extends React.SelectHTMLAttributes<HTMLSelectElement> {
+	/**
+	 * Aligns the buttons used to reorder items relative to the Select Box.
+	 */
+	buttonAlignment?: 'center' | 'end';
+
+	/**
+	 * Items to be displayed in the Select Box.
+	 */
+	items: Array<TItem>;
+
+	/**
+	 * Label to be displayed above the Select Box.
+	 */
+	label?: string;
+
+	/**
+	 * Defines whether the Select Box supports selection of multiple items.
+	 */
+	multiple?: boolean;
+
+	/**
+	 * Handler that triggers when the content of the items prop are changed caused by reordering of items.
+	 */
+	onItemsChange?: (items: Array<TItem>) => void;
+
+	/**
+	 * Handler that triggers when a new item is selected.
+	 */
+	onSelectChange: (val: Array<string>) => void;
+
+	/**
+	 *  Selected indexes, most commonly used in combination with the  Dual Listbox component
+	 */
+	selectedIndexes?: Array<number>;
+
+	/**
+	 * Amount of items that can fit inside the both Select Boxes before a scrollbar is introduced.
+	 */
+	size?: number;
+
+	/**
+	 * Defines whether the component should render buttons that allow reordering of items.
+	 */
+	showArrows?: boolean;
+
+	/**
+	 * Value of the component.
+	 */
+	value: string | Array<string>;
+
+	/**
+	 * Path to the spritemap that Icon should use when referencing symbols.
+	 */
+	spritemap?: string;
+}
+
+export const getSelectedIndexes = (
+	items: Array<TItem>,
+	selectedValues: Array<number | string>
+) =>
+	items.reduce((acc: Array<number>, item: TItem, index) => {
+		if (selectedValues.includes(item.value)) {
+			return [...acc, index];
+		}
+
+		return acc;
+	}, []);
+
+const ClaySelectBox: React.FunctionComponent<IProps> = ({
+	buttonAlignment = 'right',
+	className,
+	items,
+	label,
+	multiple,
+	onItemsChange,
+	onSelectChange,
+	showArrows,
+	size,
+	spritemap,
+	value,
+}) => {
+	const selectedIndexes = getSelectedIndexes(
+		items,
+		Array.isArray(value) ? value : [value]
+	);
+
+	return (
+		<div className={classNames(className, 'form-group')}>
+			{label && <label className="reorder-label">{label}</label>}
+
+			<div
+				className={classNames('clay-reorder', {
+					[`clay-reorder-footer-${buttonAlignment}`]: buttonAlignment,
```

Comment:
Not sure what class this outputs I broke the storybook. It should be `<div class="clay-reorder clay-reorder-footer-end">`. The other option is to add `clay-reorder-footer-end` to `clay-reorder-footer` below.